### PR TITLE
Only fire 'stream' once per pair of tracks

### PR DESCRIPTION
**Pull request by RationalCoding**
Since 'onTrack' fires the 'stream' event, MediaStreams with both audio and video will fire it twice. (See this [CodePen](https://codepen.io/RationalCoding/pen/JWPeXZ?editors=1111))

This PR stores the IDs so that 'stream' will only be fired on the first track.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -111,6 +111,7 @@ function Peer (opts) {
     }
   }
 
+  self._previousStreams = []
```
Comment (feross) [inline index.js]: This should be higher up, next to where the other private variables are declared, IMO.

File: index.js [hunk 2]
```diff
@@ -676,6 +677,9 @@ Peer.prototype._onTrack = function (event) {
   var self = this
   if (self.destroyed) return
   self._debug('on track')
+  var id = event.streams[0].id
+  if (self._previousStreams.indexOf(id) !== -1) return // Only fire one 'stream' event per track
+  self._previousStreams.push(id)
```
Comment (feross) [inline index.js]: We should set `self._previousStreams` to `null` in `destroy()`.

File: index.js [hunk 3]
```diff
@@ -676,6 +677,9 @@ Peer.prototype._onTrack = function (event) {
   var self = this
   if (self.destroyed) return
   self._debug('on track')
+  var id = event.streams[0].id
+  if (self._previousStreams.indexOf(id) !== -1) return // Only fire one 'stream' event per track
```
Comment (feross) [inline index.js]: I think this comment could be clearer. How about "Only fire one 'stream' event, even though there may be multiple tracks per stream"

**Comment by RationalCoding** [comment]
Added the fix for #153

**Comment by feross** [comment]
Released as 6.4.4

**Pull request merged**